src/lib.rs

```diff
@@ -412,57 +436,22 @@ pub struct Thread<BlockNumber, Moment, AccountId> {
     /// Category in which this thread lives
     category_id: CategoryId,
 
-    /// The thread number of this thread in its category, i.e. total number of thread added (including this)
-    /// to a category when it was added.
-    /// Is needed to give light clients assurance about getting all threads in a given range,
-    /// `created_at` is not sufficient.
-    /// Starts at 1 for first thread in category.
-    nr_in_category: u32,
-
     /// Possible moderation of this thread
     moderation: Option<ModerationAction<BlockNumber, Moment, AccountId>>,
 
-    /// Number of unmoderated and moderated posts in this thread.
-    /// The sum of these two only increases, and former is incremented
-    /// for each new post added to this thread. A new post is added
-    /// with a `nr_in_thread` equal to this sum
-    ///
-    /// When there is a moderation
-    /// of a post, the variables are incremented and decremented, respectively.
-    ///
-    /// These values are vital for light clients, in order to validate that they are
-    /// not being censored from posts in a thread.
-    num_unmoderated_posts: u32,
-    num_moderated_posts: u32,
-
     /// When thread was established.
     created_at: BlockchainTimestamp<BlockNumber, Moment>,
 
     /// Author of post.
     author_id: AccountId,
-}
 
-impl<BlockNumber, Moment, AccountId> Thread<BlockNumber, Moment, AccountId> {
-    fn num_posts_ever_created(&self) -> u32 {
-        self.num_unmoderated_posts + self.num_moderated_posts
-    }
+    /// poll
+    include_poll: bool,
```

Comment:
Why did you choose to not keep a representation of possible poll state inside of the `Thread` type? This would have been clearer to understand, and safer, as it would require at least one less runtime invariant, as you now need to have separate storage fields that have a consistent state. E.g. you now have the following invariant to maintain

`PollDataByAccount has key (id, _)  => exists t s.t. ThreadById maps id to t and t.include_poll == true`

It seems safer to have something like

```Rust
pub struct Thread<BlockNumber, Moment, AccountId> {
... 
poll: Option<Poll>
}

pub struct Poll {
 // more detailed state about the poll, such as 
// description
// timeout, 
// current cumulative tally, etc.
// voting state container <== this can be moved out, see below.
}
```

If we are concerned that there will possibly be a very large number of votes in a poll, and that this can result in having to load a large object from the runtime database when someone votes or cancels a vote, then we can do

1. put vote count field, which prevents size from growing.
2. put only the mapping with who as voted in a separate storage field, similar to what you are already doing,.

Reply:
I agree, to put an Option<Poll> in thread make it much clearer and safer than a boolean value. I will update it.